cordova-lib/src/cordova/platform.js

```diff
@@ -110,15 +110,25 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if(spec && pkgJson) {
-                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                // If there is a spec specified during add, add to pkg.json dependencies(if exists).
```

Comment:
Ahh yes. That makes sense. It only happens if the user uses `--save` and explicitly uses a version different than the one in package.json (doesn't semver.satisfy). Instead of always writing to package.json like you do below, I think you should see if `pkgJson.dependencies[prefixCordovaPlatform]` or `pkgJson.dependencies[platform]` exist and semver satisfy the passed in spec. If they do exist and satisfy fails, then write to package.json